ceci/stage.py

```diff
@@ -324,6 +663,94 @@ def read_config(self, args):
 
         return my_config
 
+
+    def iterate_fits(self, tag, hdunum, cols, chunk_rows):
+        """
+        Loop through chunks of the input data from a FITS file with the given tag
+        """
+        fits = self.open_input(tag)
+        ext = fits[hdunum]
+        n = ext.get_nrows()
+        for start,end in self.data_ranges_by_rank(n, chunk_rows):
+            data = ext.read_columns(cols, rows=range(start, end))
+            yield start, end, data
+
+    def iterate_hdf(self, tag, group_name, cols, chunk_rows):
+        """
+        Loop through chunks of the input data from an HDF5 file with the given tag.
+
+        All the selected columns must have the same length.
+
+        """
+        import numpy as np
+        hdf = self.open_input(tag)
+        group = hdf[group_name]
+
+        # Check all the columns are the same length
+        N = [len(group[col]) for col in cols]
+        n = N[0]
+        if not np.equal(N,n).all():
+            raise ValueError(f"Different columns among {cols} in file {tag}\
+            group {group_name} are different sizes - cannot use iterate_hdf")
+
+        # Iterate through the data providing chunks
+        for start, end in self.data_ranges_by_rank(n, chunk_rows):
+            data = {col: group[col][start:end] for col in cols}
+            yield start, end, data
+
+
+
+
+
+################################
+# Pipeline-related methods
+################################
+
+    @classmethod
+    def generate_command(cls, inputs, config, outputs, missing_inputs_in_outdir=False):
+        """
+        Generate a command line that will run the stage
+        """
+        module = cls.get_module()
+        module = module.split('.')[0]
+
+        flags = [cls.name, f'--config={config}']
+
+        def get_path(d, tag, ftype):
+            return fpath
+
+
+        for tag,ftype in cls.inputs:
+            if isinstance(inputs, str):
+                fn = ftype.make_name(tag)
+                fpath = f'{source}/{fn}'
```

Comment:
thanks - should be `inputs`.  It's to support just specifying an inputs directory instead of the separate paths.  This isn't exposed right now, but might be later.